test/unit/spec/VerifyCustomFactor_spec.js

```diff
@@ -0,0 +1,268 @@
+    }
+    router.refreshAuthState('dummy-token');
+    await Expect.waitForMfaVerify();
+
+    router.appState.set('lastFailedChallengeFactorData', lastFailedChallengeFactorData);
+
+    setNextResponse(resChallengeClaimsProvider);
+    router.verifyClaimsFactor();
+    await Expect.waitForVerifyCustomFactor();
+
+    const $forms = $sandbox.find('.o-form');
+
+    let forms = _.map($forms, function(form) {
+      return new VerifyCustomFactorForm($(form));
+    });
+
+    if (forms.length === 1) {
+      forms = forms[0];
+    }
+    const beacon = new Beacon($sandbox);
+
+    return {
+      router: router,
+      form: forms,
+      beacon: beacon,
+      ac: authClient,
+      setNextResponse: setNextResponse,
+      successSpy: successSpy,
+      afterErrorHandler: afterErrorHandler,
+    };
+  }
+
+  Expect.describe('Claims Provider Factor', function() {
+    Expect.describe('when features.skipIdpFactorVerificationBtn is true', function() {
+      const lastFailedChallengeFactorData = {
+        errorMessage: 'Verify failed.'
+      };
+
+      let settings = {};
+
+      beforeEach(() => {
+        settings = { 'features.skipIdpFactorVerificationBtn': true };
+      });
+
+      itp('hides the button bar', function() {
+        return setup(settings)
+          .then(function(test) {
+            Expect.isNotVisible(test.form.buttonBar());
+          });
+      });
+
+      itp('shows the button bar when lastFailedChallengeFactorData is not null', function() {
+        return setup(settings ,lastFailedChallengeFactorData).then(function(test) {
+          expect(test.form.$('.o-form-button-bar').hasClass('hide')).toBe(false);
+        });
+      });
+
+      itp('shows the waiting spinner', function() {
+        return setup(settings)
+          .then(function(test) {
+            expect(test.form.hasSpinner()).toBe(true);
+          });
+      });
+
+      itp('hides the waiting spinner when lastFailedChallengeFactorData is not null', function() {
+        return setup(settings, lastFailedChallengeFactorData).then(function(test) {
+          expect(test.form.hasSpinner()).toBe(false);
+        });
+      });
+
+      itp('shows the right auto redirect subtitle', function() {
+        return setup(settings)
+          .then(function(test) {
+            expectSubtitleToBe(test, 'Redirecting to IDP factor...');
+          });
+      });
+
+      itp('shows the right auto redirect subtitle when lastFailedChallengeFactorData is not null', function() {
+        return setup(settings, lastFailedChallengeFactorData).then(function(test) {
+          expectSubtitleToBe(test, 'Clicking below will redirect to verification with IDP factor');
+        });
+      });
+
+      itp('redirects automatically to third party', function() {
+        spyOn(SharedUtil, 'redirect');
+        return setup(settings)
+          .then(function(test) {
+            test.setNextResponse([resChallengeClaimsProvider, resSuccess]);
```

Comment:
I don't think you need this line. The `setup` function gets all the way to challenge claims provider, and the redirect happens automatically, which will call `SharedUtil.redirect` already. This line has no effect on the test